src/services/schedule.js

```diff
@@ -84,22 +85,45 @@ const ScheduleService = () => {
       return afterHoursStart && beforeHoursEnd;
     }
 
-    // Helper function to get beginning of next day
+    /*
+     *  Helper function to get beginning of next valid interval
+     * In most cases, this just means the next day, but in the
+     * case that the user doesn't allow scheduling on weekends,
+     * this would mean the following Monday.
+     */
```

Comment:
So this was the crux of the problem -- if weekend scheduling is not enabled, we need to skip to the nearest Monday rather than skip forward just one day.